packages/date-picker/test/dropdown.test.js

```diff
@@ -39,6 +40,20 @@ describe('dropdown', () => {
     });
   });
 
+  it('should restore focus to the field on outside click', async () => {
+    datepicker.focus();
+    await open(datepicker);
+    outsideClick();
+    expect(document.activeElement).to.equal(input);
+  });
+
+  it('should focus the field on outside click', async () => {
+    expect(document.activeElement).to.equal(document.body);
```

Comment:
Not sure if I clearly understand the purpose of this particular test. How is it different from the previous one?

Reply:
In this test, the focus is not in the field when the dropdown is opened.

I added this test while experimenting with `overlay.restoreFocusOnClose`: the feature does the job if the focus is in the field as the dropdown opens (overlay gets a reference to the correct node which it then re-focuses on close), but if the focus is not in the field when opened, the focus would get restored to a wrong element on close without some additional logic.

I changed the implementation since then but decided to leave the test since we may at some point in the future consider changing the implementation again.